administrator/components/com_joomgallery/helpers/imgtools.php

```diff
@@ -2323,13 +2323,13 @@ protected static function checkMemory($memory_needed)
 
       // Get memory limit in bytes
       $memory_limit = @ini_get('memory_limit');
-      if(!empty($memory_limit) && $memory_limit != 0)
+      if(!empty($memory_limit) && $memory_limit > 1)
       {
         $val          = substr($memory_limit, -1);
         $memory_limit = substr($memory_limit, 0, -1) * $byte_values[$val];
       }
 
-      if($memory_limit != 0 && $memoryNeeded > $memory_limit)
+      if($memory_limit > 1 && $memoryNeeded > $memory_limit)
       {
         $memoryNeededMB = round($memoryNeeded / $byte_values['M'], 0);
         $memoryLimitMB  = round($memory_limit / $byte_values['M'], 0);
```

Comment:
Now the only left possibility for a php warning would be id someone put an empty limit value in the php.ini file. This way $memory_limit would still be an empty string. I am not sure how php will handle that in this calculation. Thereore if propose here to change this line into
`$memoryLimitMB = round(intval($memory_limit) / $byte_values['M'], 0);`